aws-codeguruprofiler-profilinggroup/src/test/java/software/amazon/codeguruprofiler/profilinggroup/UpdateHandlerTest.java

```diff
@@ -58,23 +69,221 @@
     private final String revisionId = "TestRevisionId-123";
 
     private final GetPolicyRequest getPolicyRequest =
-        GetPolicyRequest.builder().profilingGroupName(profilingGroupName).build();
+            GetPolicyRequest.builder().profilingGroupName(profilingGroupName).build();
+
+    private final GetNotificationConfigurationRequest getNotificationConfigurationRequest =
+            GetNotificationConfigurationRequest.builder().profilingGroupName(profilingGroupName).build();
+
+    private final RemoveNotificationChannelRequest removeNotificationChannelRequest =
+            RemoveNotificationChannelRequest.builder()
+                    .profilingGroupName(profilingGroupName)
+                    .channelId("channelId")
+                    .build();
+
+    private final AddNotificationChannelsRequest addNotificationChannelsRequest = AddNotificationChannelsRequest.builder()
+            .profilingGroupName(profilingGroupName)
+            .channels(Channel.builder()
+                    .uri("channelUri2")
+                    .eventPublishers(ImmutableSet.of(EventPublisher.ANOMALY_DETECTION))
+                    .build())
+            .build();
 
     private final List<String> principals = Arrays.asList("arn:aws:iam::123456789012:role/UnitTestRole");
 
+    @Nested
+    class WhenNotificationChannelConfigurationIsProvided {
+        @BeforeEach
+        public void setup() {
+            request = makeRequest(ResourceModel.builder().profilingGroupName(profilingGroupName).anomalyDetectionNotificationConfiguration(
+                    Collections.singletonList(software.amazon.codeguruprofiler.profilinggroup.Channel.builder()
+                            .channelUri("channelUri")
+                            .build()))
+                    .build());
+
+            doReturn(GetPolicyResponse.builder().build())
+                    .when(proxy).injectCredentialsAndInvokeV2(eq(getPolicyRequest), any());
+
+
```

Comment:
Minor: Extra blank line

Reply:
This comment was marked as resolved, but not changed. I don't mind not changing the minor stuff (hence why I mark it as minor), but marking as resolved without context is a bit confusing -- did you mean to fix it but forgot? Do you think it's not worth changing it?

Without your POV on the discussion the "resolve conversation" bit doesn't tell me a lot :)

Reply:
Sorry about this, I prematurely marked these as resolved to help comb through the comments and forgot to upload a revision. Change incoming! Also the Github PR ui confuses me a bit, for instance some of the comments don't have a place to reply on unless I click on an individual file and view them from there :[